ci/glci/openstack_image.py

```diff
@@ -0,0 +1,75 @@
+import os
+import time
+import argparse
+
+from glci.model import OpenRC
+
+from openstack import connect
+
+class OpenstackImageUploader:
+  '''OpenstackImageUploader is a client to upload images to Openstack Glance.'''
+
+  def __init__(self, openrc: OpenRC):
+      self.conn = connect(
+          auth_url=openrc.auth_url,
+          project_name=openrc.project_name,
+          username=openrc.username,
+          password=openrc.password,
+          region_name=openrc.region,
+          user_domain_name=openrc.domain,
+          project_domain_name=openrc.domain,
+      )
+
+  def upload_image(self, name: str, path: str, meta: dict):
+      '''Upload an image from filesystem to Openstack Glance.'''
+
+      image = self.conn.image.create_image(
+          name=name,
+          filename=path,
+          disk_format='vmdk',
+          container_format='bare',
+          visibility='community',
+          meta=meta,
+          timeout=86400,
+      )
+      return image['id']
+
+  def wait_image_ready(self, image_id: str):
+      '''Wait until an image get in ready state.'''
+
+      while True:
+          image = self.conn.image.get_image(image_id)
+          if image['status'] == 'queued' or image['status'] == 'saving':
+              time.sleep(10)
```

Comment:
nit: add sleep-time as argument with default value?

Reply:
yep will do it.